catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/app/UserApplication.java

```diff
@@ -164,6 +211,49 @@ private Map getSubjectPreferences(Subject subject) {
     return Collections.emptyMap();
   }
 
+  private List<Map<String, Object>> getSubjectNotifications(Subject subject) {
+    String userid = subjectIdentity.getUniqueIdentifier(subject);
+    try {
+      String filter = String.format("user = '%s'", userid);
+      List<Map<String, Object>> notificationsList =
+          persistentStore.get(
+              PersistenceType.NOTIFICATION_TYPE.toString(), filter, 0, MAX_PAGE_SIZE);
+      return notificationsList.stream().map(this::mapAttributes).collect(Collectors.toList());
+    } catch (PersistenceException e) {
+      LOGGER.info(
+          "PersistenceException while trying to retrieve persisted notifications for user {}",
+          userid,
+          e);
+    }
+    return Collections.emptyList();
+  }
+
+  private Map<String, Object> mapAttributes(Map<String, Object> persistentItem) {
+    persistentItem = PersistentItem.stripSuffixes(persistentItem);
+    /*
+     *  The persistent store implementation only returns a collection if there is
+     *  more than one item in the field, else it returns a string
+     *  We need to check if it is a string, and if so add it to a collection
+     */
+    Object srcs = persistentItem.getOrDefault("src", Collections.emptySet());
```

Comment:
✏️ I think this comment is helpful. Just verify it passes the formatter. I think it'll be fine.